daemon/oci_linux.go

```diff
@@ -316,12 +319,15 @@ func setNamespaces(daemon *Daemon, s *specs.Spec, c *container.Container) error
 			nsUser.Path = fmt.Sprintf("/proc/%d/ns/user", ic.State.GetPID())
 			setNamespace(s, nsUser)
 		}
-	} else if c.HostConfig.IpcMode.IsHost() {
+	case ipcMode.IsHost():
 		oci.RemoveNamespace(s, specs.LinuxNamespaceType("ipc"))
-	} else {
+	case ipcMode.IsPrivate(), ipcMode.IsShareable():
 		ns := specs.LinuxNamespace{Type: "ipc"}
 		setNamespace(s, ns)
+	default:
```

Comment:
Right, so now I see that yet another reason for `""` appearing here is when a container is created using older version of the daemon, and then started using the new version. We definitely need to handle this case. Will fix.